dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/visitor/generator/ImportGenerator.java

```diff
@@ -38,6 +39,33 @@ public void startGeneration(int section, JavaCodeWriter writer)
 					}
 				}
 			}
+		} else if (section == CodeGenerationPhase.STATIC_SECTION && PagesVersionHandler.isPages31OrHigherLoaded()) {
+			NamedNodeMap attributes = element.getAttributes();
+			if (attributes != null) {
+				for (int i = 0; i < attributes.getLength(); i++) {
+					Node attribute = attributes.item(i);
+					String directiveName = attribute.getNodeName();
+					String directiveValue = attribute.getNodeValue();
+					if (directiveName.equals("import")) {
+						StringTokenizer tokenizer = new StringTokenizer(directiveValue, ",");
+						writeDebugStartBegin(writer);
+						while (tokenizer.hasMoreTokens()) {
+							String singleImport = ((String) tokenizer.nextToken()).trim();
+							if(singleImport.endsWith(".*")){
+								writer.println("importPackageList.add(\"" + singleImport.replace(".*", "") + "\");");
+							} else {
+								writer.println("importClassList.add(\"" + singleImport + "\");");
+							}
+
+						}
+						writeDebugStartEnd(writer);
+					}
+				}
+			}
+			// Pages 1.10 Directive Packages java.lang.*, jakarta.servlet.*, jakarta.servlet.jsp.*, and jakarta.servlet.http.* are imported implicitly by the JSP container.
+			writer.println("importPackageList.add(\"jakarta.servlet\");");
+			writer.println("importPackageList.add(\"jakarta.servlet.jsp\");");
+			writer.println("importPackageList.add(\"jakarta.servlet.http\");");
```

Comment:
This is just writing code to the servlet. This loops though the imports, and writes code which adds them to a list (in a static block) depending on whether it's a package or class.  
i.e.: 
```
    static {
    importClassList.add("io.openliberty.pages31.fat.misc.Coffee");
    importClassList.add("io.openliberty.pages31.fat.misc.Cafe");
    importPackageList.add("io.openliberty.pages31.fat.misc.other");
    importPackageList.add("jakarta.servlet");
    importPackageList.add("jakarta.servlet.jsp");
    importPackageList.add("jakarta.servlet.http");
  }
  ```